src/lib.rs

```diff
@@ -232,12 +248,19 @@ impl Watch {
             .exec()
             .context("cannot get package's metadata")?;
         let target_path = metadata.target_directory.as_std_path();
-        let build_path = &metadata.workspace_root.as_std_path().join(watch_path);
+        let build_path = &metadata.workspace_root.as_std_path().join(build_dir_path);
 
         watcher
             .watch(&metadata.workspace_root, RecursiveMode::Recursive)
             .context("cannot watch this crate")?;
 
+        for path in &self.watch_paths {
+            match watcher.watch(path, RecursiveMode::Recursive) {
+                Ok(()) => {}
+                Err(err) => log::error!("cannot watch \"{:?}\": {}", path, err),
```

Comment:
You forgot the api of std::path? xD
```suggestion
                Err(err) => log::error!("cannot watch {}: {}", path.display(), err),
```